client/src/services/auth.ts

```diff
@@ -68,3 +76,72 @@ export const handleVerifyEmail = async (
     });
   return error;
 };
+
+export const signIn = async (
+  email: string,
+  password: string
+): Promise<{ email: string; password: string }> => {
+  let errors = { email: "", password: "" };
+  if (!email.length || !password.length) {
+    errors = {
+      email: email.length ? "" : AuthErrorMessage["empty-email"],
+      password: password.length ? "" : AuthErrorMessage["empty-password"],
+    };
+  } else {
+    errors = await firebase
+      .auth()
+      .signInWithEmailAndPassword(email, password)
+      .then(async () => {
+        const result = await firebase
```

Comment:
For readability, please split up this promise chain using helper functions or return promises so that you don't need to check for errors every time.